packages/ember-simple-auth/index.d.ts

```diff
@@ -0,0 +1,140 @@
+interface Callback {
+  (): void;
+}
+type Data = Record<string, unknown>;
+interface SessionData<T> {
+  authenticated: T;
+}
+
+declare module 'ember-simple-auth/services/session' {
+  import Evented from '@ember/object/evented';
+  import Service from '@ember/service';
+  import type BaseSessionStore from 'ember-simple-auth/session-stores/base';
+  import type Transition from '@ember/routing/-private/transition';
+
+  export default class SessionService<T> extends Service.extend(Evented) {
+    isAuthenticated: boolean;
+    data: SessionData<T>;
+    store: BaseSessionStore;
+    attemptedTransition: Transition | null;
+
+    authenticate(authenticator: string, ...args: unknown[]): Promise<void>;
+    invalidate(...args: unknown[]): Promise<unknown>;
+    requireAuthentication(transition: Transition, routeOrCallback: string | Callback): boolean;
+    prohibitAuthentication(routeOrCallback: string | Callback): boolean;
+    handleAuthentication(routeAfterAuthentication: string): void;
+    handleInvalidation(routeAfterInvalidation: string): void;
+    setup(): Promise<void>;
+  }
+}
+
+declare module 'ember-simple-auth/authenticators/base' {
+  import EmberObject from '@ember/object';
+  import Evented from '@ember/object/evented';
+
+  export default class extends EmberObject.extend(Evented) {
+    restore(data: Data): Promise<unknown>;
+    authenticate(...args: unknown[]): Promise<unknown>;
```

Comment:
I believe `Promise<unknown>` here can actually be `Promise<SessionData<T>>`.
I don't know how to make sure `SessionService<T>.data` and these return values are in synced though.